nsqd/nsqd.go

```diff
@@ -370,6 +379,11 @@ func (n *NSQD) PersistMetadata() error {
 		return err
 	}
 
+	err = os.Remove(oldFileName)
```

Comment:
My concern with this approach is that it breaks the ability to roll back from a nsqd upgrade (old nsqd would not see new file). 

As much as i hate things sticking around, I'd rather see a phased deprecation where one release persists both versions with defined precedence on startup (or requiring equality between two files), and a subsequent release remove the old file.